### PR TITLE
Fix #168 - Ctest shall output failures

### LINKED ISSUE #168: ctest --output-on-failure
It would be nice, for CI, but also for regular development, to see the actual error messages when there are errors. Apparently, this can be achieved by passing `--output-on-failure` as argument to ctest, or setting `CTEST_OUTPUT_ON_FAILURE=plouf` in the environment.

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Fix #168

### CODE REVIEW

**Comment by VincentRouvreau** [comment]
@mglisse could you please tell me if you are ok with this PR as some PR have failed tests and we cannot investigate on ?

File: src/common/doc/installation.h [hunk 1]
```diff
@@ -32,7 +32,10 @@ make \endverbatim
  *
  * \subsection testsuites Test suites
  * To test your build, run the following command in a terminal:
- * \verbatim  make test \endverbatim
+ * \verbatim make test \endverbatim
+ * `make test` is using <a href="https://cmake.org/cmake/help/latest/manual/ctest.1.html">Ctest<\a> (CMake test driver
+ * program). If some of the tests are failing, plend send us the result of the following command:
```
Comment (mglisse) [inline src/common/doc/installation.h]:
```suggestion
 * program). If some of the tests are failing, please send us the result of the following command:
```

File: src/python/doc/installation.rst [hunk 2]
```diff
@@ -98,6 +98,18 @@ following command in a terminal:
     export PYTHONPATH='$PYTHONPATH:/path-to-gudhi/build/python'
     make test
 
+`make test` is using
+`Ctest <https://cmake.org/cmake/help/latest/manual/ctest.1.html>`_ (CMake test
+driver program). If some of the tests are failing, plend send us the result of
```
Comment (mglisse) [inline src/python/doc/installation.rst]:
```suggestion
driver program). If some of the tests are failing, please send us the result of
```

File: src/python/doc/installation.rst [hunk 3]
```diff
@@ -98,6 +98,18 @@ following command in a terminal:
     export PYTHONPATH='$PYTHONPATH:/path-to-gudhi/build/python'
     make test
```
Comment (mglisse) [inline src/python/doc/installation.rst]: Shouldn't we directly say to run `ctest` (possibly suggesting `-j8`, `--output-on-failure` as optional options)?

Comment (VincentRouvreau) [reply src/python/doc/installation.rst]: Done on 38f0ff98cf6da1e882b6b44fc3ed7b3d310fb91f (+ conflict resolution cec28c5ee04886ff30d4f3960d713e044837ff4f )

**Comment by VincentRouvreau** [comment]
> I don't really see the relation between the coverage / boost-test change and the issue, but I have no objection to it.

All the C++ unitary tests were using a function called `gudhi_add_coverage_test` because by default it was compiling with `-pg  -fprofile-arcs -ftest-coverage` compilation options and was generating an xml file, summary of tests and coverage reports, for Jenkins purpose. This was the default behaviour.
Now, we want the failing tests to be verbose for the developer by default. This why I renamed the function `gudhi_add_boost_test` and I kept the coverage mechanism because it can help in code reviews.

> It is always possible to run the tests locally.

We agree.

**Pull request merged**